.github/actions/veracode/action.yml

```diff
@@ -4,20 +4,35 @@ inputs:
   srcclr-api-token:
     description: "Agent API Token"
     required: true
-  srcclr-install-default-options:
-    description: Default options for Source Clear scan
-    required: false
-    default: '-DskipTests -Dmaven.javadoc.skip=true'
-  srcclr-install-options:
-    description: Additional options for Source Clear scan
+  srcclr-project-ext:
+    description: "Direct scan results to Veracode project named: <default project name>/<srcclr-project-ext>"
     required: false
 runs:
   using: "composite"
   steps:
+    - name: "Install srcclr CLI tool"
+      shell: bash
+      run: |
+        # Install srcclr CLI tool (Veracode) as per https://docs.veracode.com/r/t_sc_cli_agent
+        curl -sSL 'https://keyserver.ubuntu.com/pks/lookup?op=get&search=0xdf7dd7a50b746dd4' | sudo gpg --dearmor -o /usr/share/keyrings/veracode-sca-archive.gpg
+        echo 'deb [signed-by=/usr/share/keyrings/veracode-sca-archive.gpg] https://download.sourceclear.com/ubuntu stable/' | sudo tee /etc/apt/sources.list.d/veracode-sca.list
+        sudo apt-get update
+        sudo apt-get install srcclr
+    - name: "Set target project URI, with extension if provided"
+      shell: bash
+      run: |
+        SRCCLR_SCM_URI="${{ github.server_url }}/${{ github.repository }}"
+
+        if [ -n "${{ inputs.srcclr-project-ext }}" ]; then
+          SRCCLR_SCM_URI="$SRCCLR_SCM_URI/${{ inputs.srcclr-project-ext }}"
+        fi
+
+        echo "SRCCLR_SCM_URI=$SRCCLR_SCM_URI" >> $GITHUB_ENV
     - name: "Run scanning"
       run: ${{ github.action_path }}/source_clear.sh
       shell: bash
       env:
         SRCCLR_API_TOKEN: ${{ inputs.srcclr-api-token }}
-        SRCCLR_INSTALL_DEFAULT_OPTIONS: ${{ inputs.srcclr-install-default-options }}
-        SRCCLR_INSTALL_OPTIONS: ${{ inputs.srcclr-install-options }}
+        SRCCLR_SCM_REF: ${{ github.event_name == 'pull_request' && github.ref || github.ref_name }}
+        SRCCLR_SCM_REF_TYPE: ${{ github.event_name == 'pull_request' && 'tag' || 'branch' }}
```

Comment:
There are ways to fetch the original branch name from a PR event if needed, using `github.head_ref` instead of `github.ref`!

Also, I'm not sure I understand the logic related to PR = tag and rest = branch?

Reply:
I know it is possible, but I don't think that's what we want. 

A PR is a combination of the current master + changes proposed, a branch has the old master base and new changes proposed. I don't want to mix the 2.

If I target a branch instead of a tag, I will be showing potentially appeared/removed issues from new base as if it was on the old base. Gets a bit confusing.
Might as well use the tag then, it works fine, shows vulnerabilities properly in Veracode scans and doesn't confuse the base branch results with itself, especially given that current PR SCA scans just throw 

`Error:  Failed to execute goal com.srcclr:srcclr-maven-plugin:3.1.25:scan (default-cli) on project alfresco-community-repo: Could not perform scan because an apiToken was not provided. Please provide it either through the "apiToken" Maven parameter or the "SRCCLR_API_TOKEN" environment variable -> [Help 1]`

Reply:
Alright, sounds like you planned this appropriately then. LGTM!